File: vipclaw-admin/src/main/java/com/vipclaw/admin/service/AuthService.java
package com.vipclaw.admin.service;

import com.vipclaw.admin.dto.LoginRequest;
import com.vipclaw.admin.dto.LoginResponse;

/**
 * 认证服务接口
 */
public interface AuthService {

    /**
     * 用户登录
     * @param request 登录请求
     * @return 登录响应
     */
    LoginResponse login(LoginRequest request);

    /**
     * 获取当前登录用户信息
     * @return 用户信息
     */
    LoginResponse.UserInfo getCurrentUser();

    /**
     * 退出登录
     */
    void logout();
}

File: harnax-admin/src/main/kotlin/com/agnetix/harnax/admin/service/AuthService.kt
package com.agnetix.harnax.admin.service

import com.agnetix.harnax.admin.dto.LoginRequest
import com.agnetix.harnax.admin.dto.LoginResponse

/**
 * Authentication service interface
 */
interface AuthService {

    /**
     * User login
     * @param request Login request
     * @return Login response
     */
    fun login(request: LoginRequest): LoginResponse

    /**
     * Mobile login (no captcha required)
     * @param request Login request with SHA-256 hashed password
     * @return Login response
     */
    fun mobileLogin(request: LoginRequest): LoginResponse

    /**
     * Logout
     */
    fun logout()
}
